packages/backend/src/projectAdapters/baseProjectAdapter.ts

```diff
@@ -0,0 +1,15 @@
+import {Explore} from "common";
+
+export class BaseProjectAdapter {
+    constructor() {
+    }
+
+    public async compileAllExplores(): Promise<Explore[]> {
+        throw new Error('Not implemented');
+    }
+
+    public async runQuery(sql: string): Promise<Record<string, any>> {
+        throw new Error('Not implemented');
+    }
+
+}
```

Comment:
We could use an interface instead of this base class. 
The advantages of using an interface is that it will error on build if they don't implement some of the methods. With the class we only get the error at run time.

```
interface ProjectAdapter {
    compileAllExplores(): Promise<Explore[]>;
    runQuery(sql: string): Promise<Record<string, any>>;
}
```

Implement interface

```
export class DbtBaseProjectAdapter implements ProjectAdapter {
```

Reply:
Much better thanks